Zend/Optimizer/zend_inference.c

```diff
@@ -3444,16 +3444,12 @@ static zend_always_inline int _zend_update_type_info(
 						} else if (ssa_op->op1_use >= 0 && !ssa->var_info[ssa_op->op1_use].is_instanceof) {
 							ce = ssa->var_info[ssa_op->op1_use].ce;
 						}
-						if (prop_info) {
```

Comment:
If the problem is related to typed properties only, can we add a check for ``prop_info->type`` here?

Reply:
I checked, the problem is related not to typed property but any unset property. So the patch looks right.

Reply:
Yeah, sorry the comment is misleading. I'll adjust it.